src/sentry/rules/conditions/first_seen_event.py

```diff
@@ -13,3 +18,16 @@ def passes(self, event: Event, state: EventState) -> bool:
             return state.is_new
         else:
             return state.is_new_group_environment
+
+    def get_activity(self, start: datetime, end: datetime) -> Sequence[Activity]:
+        return [
+            Activity(
+                project=g.project,
+                group=g,
+                type=ActivityType.FIRST_SEEN.value,
+                datetime=g.first_seen,
+            )
+            for g in Group.objects.filter(
+                project=self.project, first_seen__gte=start, first_seen__lt=end
+            )
+        ]
```

Comment:
This could potentially return an extremely large set of data. I'd recommend that we add an upper bound to the query, 1k-10k, not sure which, but you can experiment to see if it's particularly slow.

I'd also recommend you only fetch the data you need. So rather than fetching the entire group, just fetch `first_seen, id, project_id` maybe. Not totally sure which fields you need, but it's a lot less memory intensive to fetch a few thousand datetimes and ints than it is to fetch models. You can do this via `values_list`